src/node_task_runner.cc

```diff
@@ -252,7 +253,25 @@ FindPackageJson(const std::filesystem::path& cwd) {
 void RunTask(const std::shared_ptr<InitializationResultImpl>& result,
              std::string_view command_id,
              const std::vector<std::string_view>& positional_args) {
-  auto cwd = std::filesystem::current_path();
+  auto run_from = per_process::cli_options->run_from;
+  std::filesystem::path cwd;
+
+  if (run_from.empty()) {
+    cwd = std::filesystem::current_path();
+  } else {
+    cwd = std::filesystem::absolute(std::filesystem::path(run_from));
+
+    if (is_regular_file(cwd)) {
```

Comment:
npm's behavior only accepts passing a dir.

```sh
❯ npm --prefix ./coolpackage.json run build
npm error code ENOTDIR
npm error syscall open
npm error path /home/cwunder/GitHub/nodejs.org/coolpackage.json/package.json
```

```sh
❯ npm --prefix ./package.json run build
npm error code ENOTDIR
npm error syscall open
npm error path /home/cwunder/GitHub/nodejs.org/package.json/package.json
```

I see the code only grabs the parent_directory from the given `cwd`, so even if I pass `/some/path/to/a/coolpackage.json` it is actually going to read a package.jnson from the folder `/some/path/to/a`

Should we then at least handle a WARN or disallow passing a filename that is not a package.json file or simply only handle folders?

Reply:
I think we should not support a path to a file. To me it only makes the feature more difficult to grasp.

Reply:
I agree. @flakey5 could you please make these changes?